server/app/api/v1/charity/raffle/controller.js

```diff
@@ -0,0 +1,218 @@
+import findIndex from 'lodash/findIndex';
+import merge from 'lodash/merge';
+import { NotFoundError, NotPermittedError, ValidationError } from 'common-errors';
+import Charity from '../model';
+import Raffle from './model';
+import User from '../../user/model';
+
+import { VALID_OBJECT_ID } from '../../../../util/validations';
+
+
+/* *********************
+ * EXPORTED FUNCTIONS *
+ * *********************/
+
+/**
+ * Params middleware for any route with /raffle/:id
+ * sets req.raffle
+ */
+const params = (req, res, next, id) => {
+  Raffle.findById(id)
+    .select('-charity')
+    .populate('winner', 'email firstName lastName')
+    .exec()
+    .then((raffle) => {
+      if (!raffle) {
+        throw new NotFoundError('Raffle not found');
+      } else {
+        req.raffle = raffle;
+        next();
+      }
+    })
+    .catch(err => next(err));
+};
+
+/**
+ * Returns all raffles
+ * takes optional query isActive=true
+ */
+const all = (req, res, next) => {
+  if (req.query.hasOwnProperty('isActive') && req.query.isActive === 'true') {
+    return Raffle.find({ isActive: true })
+      .select('-charity')
+      .populate('winner', 'email firstName lastName')
+      .exec()
+      .then(raffles => res.json(raffles))
+      .catch(err => next(err));
+  }
+  return Charity
+    .populate(req.charity, {
+      path: 'raffles',
+      select: '-charity'
+    })
+    .then(({ raffles }) => {
+      return Raffle.populate(raffles, {
+        path: 'winner'
+      });
+    })
+    .then(raffles => res.json(raffles))
+    .catch(err => next(err));
+};
```

Comment:
Is it common Node practice to have this much work done in the controller? Seems like model code to me.